### PR TITLE
Add to upgrade Algolia from version 1 to version 2.

### PR DESCRIPTION
**Pull request by abdirisaq**

### CODE REVIEW

File: src/Plugin/search_api/backend/SearchApiAlgoliaBackend.php [hunk 1]
```diff
@@ -165,7 +166,7 @@ public function removeIndex($index) {
    * {@inheritdoc}
    */
   public function indexItems(IndexInterface $index, array $items) {
-    $this->connect($index);
```
Comment (toin0u) [inline src/Plugin/search_api/backend/SearchApiAlgoliaBackend.php]: You don't need the $index anymore?

Comment (abdirisaq) [reply src/Plugin/search_api/backend/SearchApiAlgoliaBackend.php]: @toin0u No, we don't need to set index inside connect method, but we saved index objects in line 185.

Comment (toin0u) [reply src/Plugin/search_api/backend/SearchApiAlgoliaBackend.php]: Ok I see - the API changed 👍

Comment (toin0u) [reply src/Plugin/search_api/backend/SearchApiAlgoliaBackend.php]: Actually not - you've refactored it. ok 👍

**Pull request merged**